### PR TITLE
1414426: Ability to get future pools by owner

### PR DESCRIPTION
**Pull request by wottop**
Either future only or active and future together.

To test this function you will need to use the two new query parameters 'add_future' and 'only_future' in conjunction with the 'activeon' parameter.

'activeon' determines what 'now' is for the other parameters. When not passed in, the current date will be used. If add_future is set to true, then you will get back any pool whose starting date is after the 'activeon' date in addition to those that the method previously returned. If only_future is set to true, then you will only get back the future date pools.

You may not use both add_future and only_future at the same time. It will return a bad request exception.

### CODE REVIEW

**Comment by barnabycourt** [comment]
Please include test instructions in the PR description.

File: server/src/main/java/org/candlepin/controller/PoolManager.java [hunk 1]
```diff
@@ -190,7 +190,8 @@ Entitlement adjustEntitlementQuantity(Consumer consumer, Entitlement entitlement
      */
     Page<List<Pool>> listAvailableEntitlementPools(Consumer consumer, ActivationKey key,
         Owner owner, String productId, String subscriptionId, Date activeOn, boolean activeOnly,
-        boolean includeWarnings, PoolFilterBuilder filterBuilder, PageRequest pageRequest);
+        boolean includeWarnings, PoolFilterBuilder filterBuilder, PageRequest pageRequest,
```
Comment (barnabycourt) [inline server/src/main/java/org/candlepin/controller/PoolManager.java]: if activeOnly is never set to False, then this parameter should be stripped out from the entire stack.

**Comment by wottop** [comment]
buildbot: retest this please

**Comment by barnabycourt** [comment]
If add_future and only_future are exclusionary, would it make sense to use an enumerated value?
entitlements_included=[current, current_and_future, future_only]?

**Comment by wottop** [comment]
It got pushed into 51 already with the separate variables.

**Pull request merged**